pytorch_transformers/tokenization_utils.py

```diff
@@ -518,6 +518,36 @@ def add_tokens(self, new_tokens):
 
         return len(to_add_tokens)
 
+    def num_added_tokens(self, pair=False):
+        """
+        Returns the number of added tokens when encoding a sequence with special tokens.
+
+        Note:
+            This encodes inputs and checks the number of added tokens, and is therefore not efficient. Do not put this
+            inside your training loop.
+
+        Args:
+            pair: Returns the number of added tokens in the case of a sequence pair if set to True, returns the
+                number of added tokens in the case of a single sequence if set to False.
+
+        Returns:
+            Number of tokens added to sequences
+        """
+
+        if pair:
+            initial_tokens_len = len(self.encode("This is a sequence") + self.encode("This is another"))
+            final_tokens = self.encode("This is a sequence", "This is another", add_special_tokens=True)
+
+            # In some models (e.g. GPT-2), there is no sequence pair encoding.
+            if len(final_tokens) == 2:
```

Comment:
Because I don't like this `len(final_tokens) == 2` test which is strange and makes no sense for the user/reader of the library.

Reply:
You're totally right, this indeed does not make sense. It is an artifact from the previous `encode` function on which I was working. Since the tokens are now concatenated in models that do not handle sequence pairs, this test will never pass. I'm removing it.